script.js

```diff
@@ -6,4 +9,317 @@ function openNav(){
+    {
+        var question = allQuestions[i].trim();
+        var currText = $("#pitScoutingTextBox").val();
+        $("#pitScoutingTextBox").val(currText + question + '\n');
+    }
+}
+
+function getElementName(str)
+{
+    // /season_config/season/elementName=elementValue
+    return str.substring(str.lastIndexOf('/')+1,str.indexOf('='));
+}
+
+function getElementValue(str)
+{
+    return str.split('=')[1].toString();
+}
+
+//updates table based on cookies from checked season
+function setTable(seasonChecked)
+{
+    var cookieList = document.cookie.split(';');
+    for (var i = 0; i < cookieList.length; i++)
+    {
+        if (cookieList[i].trim().startsWith('/season_config/' + seasonChecked))
+        {
+            var elementName = getElementName(cookieList[i]);
+            var elementPoints = getElementValue(cookieList[i]);
+            console.log('elementName: ' + elementName);
+            console.log('elementVal: ' + elementPoints);
+
+            var newTRHtml = "<tr><td>" + elementName + 
+            "</td><td>" + elementPoints + "</td></tr>";
+
+            if ($('#scoringTable').html().indexOf(newTRHtml) < 0)
+            {
+                $('#scoringTable').append(newTRHtml);
+            }
+        } else
+        {
+            $('#scoringTable').val(" ");
+        }
+    }
+}
+
+
+$(document).ready(function(){ 
+    //sets current checked season of the radiolist in a cookie
+    $.fn.setCurrCheckedSeason = function(){
+        var currCheckedSeason = this.val();
+        if (typeof(currCheckedSeason) !== "undefined")
+        {
+            document.cookie = "currCheckedSeason="+currCheckedSeason+
+            "; expires=Wed, 1 Jan 2038 12:00:00 UTC; path=/";
+        }
+    };
+
+    function getCurrCheckedSeason($param) {
+        var currCheckedSeason = $param.val();
+        if (typeof(currCheckedSeason) !== "undefined")
+        {
+            return currCheckedSeason;
+        }
+    }
+
+    //if hits enter during season text box, trigger button
+    $('#newSeasonInputBox').keypress(function(e){
+        if (e.keyCode == 13)
+        {
+            addNewSeason();
+        }
+    });
+
+    //if hits enter while entering points, trigger save button
+    $('#elementPoints').keypress(function(e){
+        if (e.keyCode == 13)
+        {
+            $('#scoringSaveButton').click();
+        }
+    });
+
+    //adds new element to table, handles things like element already exists and season not checked
+    $("#scoringSaveButton").click(function(){
+        var newElementName = $("#elementKey").val().trim().toString();
+        var newElementPoints = $("#elementPoints").val().trim();
+        
+        if (newElementName !== "" || newElementPoints !== "")
+        {
+            var season = $("input[name=seasonListItem]:checked","#seasonRadioList").val();
+            if (typeof(season) !== "undefined")
+            {
+                var cookieName = '/season_config/' + season + '/' + newElementName;
```

Comment:
you're doing this a lot - maybe make a helper method?

```JS
function season_config_cookie_name(season, element) {
  return `/season_config/${ season }/${ element }`;
}
```

This way, if you decide to tweak where the cookie is (say, `season_config` to `season_info`), but you used this helper method everywhere you either read or set a corresponding cookie, you can just change the function.

Reply:
That's neat, basically javascript's version of the java printf